### PR TITLE
Improve BayesianOptimization Oracle

### PR DESCRIPTION
**Pull request by omalleyt12**
Spoke with @sagipe , based on his suggestions this PR:

* Sets the default for `num_initial_points` to 3 times the dimensionality of the space
* Runs 20 restarts of the optimization process for the `GaussianProcessRegressor`
* Uses the `Matern` kernel for the `GaussianProcessRegressor`
* Uses 50 restarts for `scipy.minimize`. Sagi recommended this be much higher (1000s), but increasing this really slowed down suggestions so just doubled from 25 to 50 for now. We can increase this or maybe make this configurable in the future
* Scaled and normalized data so that x-values are within [0, 1], and so that log and reverse-log sampling is respected (this leads to more accurate optimization for the learning-rate, for instance)
* For distributed mode: the results of ongoing trials are "hallucinated" based on the predicted values from the last iteration of the GPR. This discourages multiple workers from trying the same params

### CODE REVIEW

File: kerastuner/tuners/bayesian.py [hunk 1]
```diff
@@ -52,7 +53,7 @@ class BayesianOptimizationOracle(oracle_module.Oracle):
     def __init__(self,
                  objective,
                  max_trials,
-                 num_initial_points=2,
+                 num_initial_points=None,
```
Comment (fchollet) [inline kerastuner/tuners/bayesian.py]: This needs to be marked as optional in the docstring, with an explanation of what default value gets used

File: kerastuner/tuners/bayesian.py [hunk 2]
```diff
@@ -73,20 +74,34 @@ def __init__(self,
...
     def _populate_space(self, trial_id):
         # Generate enough samples before training Gaussian process.
         completed_trials = [t for t in self.trials.values()
                             if t.status == 'COMPLETED']
-        if len(self.trials) < self.num_initial_points or len(completed_trials) < 2:
+
+        # Use 3 times the dimensionality of the space as the default number of
+        # random points.
+        dimensions = len(self.hyperparameters.space)
+        num_initial_points = self.num_initial_points or 3 * dimensions
```
Comment (fchollet) [inline kerastuner/tuners/bayesian.py]: This should be mentioned in the docstring

File: kerastuner/tuners/bayesian.py (hunk 1 above)
Comment (omalleyt12) [reply kerastuner/tuners/bayesian.py]: Done

File: kerastuner/tuners/bayesian.py (hunk 2 above)
Comment (omalleyt12) [reply kerastuner/tuners/bayesian.py]: Done

File: kerastuner/tuners/bayesian.py [hunk 3]
```diff
@@ -190,22 +199,30 @@ def _vectorize_trials(self):
...
-                elif isinstance(hp, hp_module.Choice):
-                    val = hp.values.index(trial_value)
-                else:
-                    val = trial_value
-                vector.append(val)
+                # Embed an HP value into the continuous space [0, 1].
+                prob = hp_module.value_to_cumulative_prob(trial_value, hp)
+                vector.append(prob)
+
+            if trial in ongoing_trials:
```
Comment (Ark-kun) [inline kerastuner/tuners/bayesian.py:206]: The `ongoing_trials` dictionary is keyed by the tuner_id, not trial.
Are there unit tests for this method?

**Pull request merged**